WalletWasabi/Crypto/ZeroKnowledge/LinearRelation/Equation.cs

```diff
@@ -0,0 +1,61 @@
+using NBitcoin.Secp256k1;
+using System.Linq;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge.LinearRelation
+{
+	// Each proof of a linear relation consists of multiple knowledge of
+	// representation equations, all sharing a single witness comprised of several
+	// secrets.
+	//
+	// Note that some of the generators can be the point at infinity, when a term
+	// in the witness does not play a part in the representation of a specific
+	// point.
+	public class Equation
+	{
+		public Equation(GroupElement publicPoint, GroupElementVector generators)
+		{
+			Guard.NotNullOrEmpty(nameof(generators), generators); // generators can be null
```

Comment:
```suggestion
			Guard.NotNullOrEmpty(nameof(generators), generators); // generators cannot be null
```

?

Also, https://github.com/zkSNACKs/WalletWasabi/issues/4241 mentions:

> Try to remove Guards from everywhere.

I would not add new `Guard.XXX` calls as NRT conventions specify that already as you did not used `GroupElementVector? generators`.

Reply:
Good catch. That comment is wrong. I will remove it. This guard is not about null but about empty instead.

Reply:
i think i'd rather just delete that comment since it adds nothing, i suspect it originated before we had the vector type refactoring, and i may have had a loop that checked each generator isn't null